lib/mini_repo/repository.ex

```diff
@@ -0,0 +1,138 @@
+defmodule MiniRepo.Repository do
+  alias MiniRepo.Repository.State
+
+  @name __MODULE__
```

Comment:
Why do we sometimes use *@name* and sometimes use *__MODULE__* I think it isn't explicit enough.